README.md

```diff
@@ -27,6 +30,33 @@ Currently the following variables are supported:
   functionality for privilege escalation, then this is the name of the target
   user to change to.
 
+### Server settings
+
+* `register_idm_domain` - Default: ipa.example.com. The IDM domain to be
+  registered with.
+* `register_idm_server_realm` - Default: upper-case transformation of
+  `register_idm_domain`.
+* `register_idm_server_domain` - Default: copy of `register_idm_domain`.
+* `reigster_idm_server_admin_username` - Default 'admin'
+* `register_idm_server_admin_password` - Default ADMIN\_PASS
```

Comment:
Does it make sense for these to not have defaults? Based on the tasks, they look like they're required, so I'm wondering if it makes for sense for the role to explode if any of these aren't set.

Reply:
Taking from our [meta_standards](https://github.com/oasis-roles/meta_standards):

```
Every argument accepted from outside of the role should be given a default 
value in defaults/main.yml. This allows a single place for users to look to 
see what inputs are expected. Document this file copiously
```

Obviously we're not doing the last part of that any longer, now that we have README.md to document the arguments (note to self: change this). However, we do mandate there be a default value so you don't get undefined variable errors. The role will still definitely explode without it, similarly to how the Travis CI run explodes. It gives, though, a convenient place to check to make sure our documentation is still up to snuff.

Reply:
Yeah, I loophole that one by putting commented out var definitions in defaults/main that say the value intentionally defaults to being undefined because the var is required for the role to function. It's worth revisiting that standard, I think.

Reply:
I've learned that YAML has the syntax of `~` for an actual Null/null/None value. So we can always go with that, if we really want to.

Reply:
Setting no value, the literal "null", "NULL", and "Null" are also accepted for this, so all of these are the same thing:
```yaml
# null_var - undefined
null_var:
null_var: # null but with a convenient comment
null_var: null
null_var: Null
null_var: NULL
null_var: ~
```

Of these, similar to preferring "true/false" over the likes of "yes/no" or "on/off", I think setting to "null" is my preference.

https://yaml.org/spec/1.2/spec.html#id2805071

Reply:
Hmm, I was leaning towards `~`, myself. But digging further I see that was the preferred syntax in YAML 1.1 and now in 1.2 it is open. I also prefer `null` and will work in that direction in the future.

Reply:
I've added "null" values, and tossed in an assertion at the top of the role that those values are not `none`.

Because `null` is the value in YAML but `none` is the value in Jinja2 and that's why we can't have nice things.

Reply:
Just so we're clear about this...

YAML: `null`
Jinja2: `none`
Python: `None`

Sound about right? :)

Reply:
> Just so we're clear about this...
> 
> YAML: `null`
> Jinja2: `none`
> Python: `None`
> 
> Sound about right? :)

Spot on!